### PR TITLE
Support signature help for partially-filled-in type arguments `f<`

### PR DESCRIPTION
**Pull request by andy-ms**
Supports signature help at `f<`, `f<T,` `f<T, U,`, etc.
Required some reworking since we no longer always have a `CallLikeExpression` handy. We now no longer expose as much to `completions.ts`, making it call a special function `getArgumentInfoForCompletions`.

### CODE REVIEW

File: src/services/signatureHelp.ts [hunk 1]
```diff
@@ -178,16 +198,35 @@ namespace ts.SignatureHelp {
...
                 kind: ArgumentListKind.JSXAttributesArguments,
-                invocation: node.parent,
+                invocation: { kind: InvocationKind.Call, node: node.parent },
                 argumentsSpan: createTextSpan(attributeSpanStart, attributeSpanEnd - attributeSpanStart),
                 argumentIndex: 0,
                 argumentCount: 1
             };
         }
-
+        else if (isBinaryExpression(node.parent)) {
```
Comment (sheetalkamat) [inline src/services/signatureHelp.ts]: Can you please use isPossiblyTypeArgumentPosition instead?

File: src/services/signatureHelp.ts [hunk 2]
```diff
@@ -32,32 +35,38 @@ namespace ts.SignatureHelp {
...
-        return typeChecker.runWithCancellationToken(cancellationToken, typeChecker => createSignatureHelpItems(candidates, resolvedSignature, argumentInfo, typeChecker));
+        return typeChecker.runWithCancellationToken(cancellationToken, typeChecker => createSignatureHelpItems(candidateInfo.candidates, candidateInfo.resolvedSignature, argumentInfo, sourceFile, typeChecker));
     }
 
-    function createJavaScriptSignatureHelpItems(argumentInfo: ArgumentListInfo, program: Program, cancellationToken: CancellationToken): SignatureHelpItems {
-        if (argumentInfo.invocation.kind !== SyntaxKind.CallExpression) {
-            return undefined;
+    function getCandidateInfo(argumentInfo: ArgumentListInfo, checker: TypeChecker): { readonly candidates: ReadonlyArray<Signature>, readonly resolvedSignature: Signature } | undefined {
+        const { invocation } = argumentInfo;
+        if (invocation.kind === InvocationKind.Call) {
```
Comment (sheetalkamat) [inline src/services/signatureHelp.ts]: what about new expressions?

**Comment by andy-ms** [comment]
Note that this feature means editors should request signature help on `<`. @amcasey @mjbvz

**Comment by mjbvz** [comment]
I've added `<` as a trigger character in VS Code: https://github.com/Microsoft/vscode/commit/3664b7c10efdf76a93dc547bd49b2fa09e6a9fda

should be in the next insiders build

**Comment by andy-ms** [comment]
Don't know why VSTS isn't running tests.

**Pull request merged**